crates/core/src/pipeline/runner.rs

```diff
@@ -108,9 +119,181 @@ impl PipelineRunner {
             fps_last_frame_ctr: 0,
             info: None,
             thumb_interval: 1800, // Disable thumbnails by default for performance
+            idle_mode: false,
+            idle_start_time: None,
+            placeholder_frame_count: 0,
         })
     }
 
+    /// Generate a placeholder black video frame
+    unsafe fn generate_placeholder_video_frame(&mut self, variant: &VideoVariant, stream_time_base: (i32, i32)) -> Result<*mut ffmpeg_rs_raw::ffmpeg_sys_the_third::AVFrame> {
+        use ffmpeg_rs_raw::ffmpeg_sys_the_third::{av_frame_alloc, av_frame_get_buffer, AVFrame, AVPixelFormat};
+        
+        let frame = av_frame_alloc();
+        if frame.is_null() {
+            bail!("Failed to allocate placeholder video frame");
+        }
+
+        (*frame).format = AVPixelFormat::AV_PIX_FMT_YUV420P as i32;
+        (*frame).width = variant.width as i32;
+        (*frame).height = variant.height as i32;
+        (*frame).time_base.num = stream_time_base.0;
+        (*frame).time_base.den = stream_time_base.1;
+        
+        // Set PTS based on frame rate and placeholder frame count
+        let fps = if variant.fps > 0.0 { variant.fps } else { 30.0 };
+        let time_base_f64 = stream_time_base.0 as f64 / stream_time_base.1 as f64;
+        (*frame).pts = (self.placeholder_frame_count as f64 / fps / time_base_f64) as i64;
```

Comment:
use `self.frame_ctr` we want to skip any data that they didnt send if they re-connect